packages/upload-client/src/blob.js

```diff
@@ -303,7 +304,38 @@ export async function add(
     })
   }
 
-  return multihash
+  // Ensure the blob has been accepted
+  const acceptReceipt = await new ReceiptPoller(options).poll(
+    nextTasks.accept.task.link()
+  )
+
+  const blocks = new Map(
+    [...acceptReceipt.iterateIPLDBlocks()].map((block) => [
+      `${block.cid}`,
+      block,
+    ])
+  )
+  const site = Delegation.view(
+    {
+      root: /** @type {import('@ucanto/interface').UCANLink} */ (
+        // @ts-ignore Property
+        acceptReceipt.out.ok.site
+      ),
+      blocks,
+    },
+    null
+  )
+  /* c8 ignore next 5 */
+  if (!site) {
```

Comment:
Why would this be falsey?

Reply:
Oh you need to not pass `null` above (pass `undefined` or simply omit) for `view(...)` to throw in the case where the root does not exist in the passed blocks.

Reply:
addressed